tidy3d/plugins/smatrix/component_modelers/terminal.py

```diff
@@ -399,14 +434,132 @@ def _set_port_data_array_attributes(data_array: PortDataArray) -> PortDataArray:
+        """Calculate antenna parameters using superposition of fields from multiple port excitations.
+
+        The method computes the radiated far fields and port excitation power wave amplitudes
+        for a superposition of port excitations, which can be used to analyze antenna radiation
+        characteristics.
+
+        Parameters
+        ----------
+        port_amplitudes : dict[str, complex] = None
+            Dictionary mapping port names to their desired excitation amplitudes. For each port,
+            :math:`\frac{1}{2}|a|^2` represents the incident power from that port into the system.
+            If None, uses only the first port without any scaling of the raw simulation data.
+        monitor_name : str = None
+            Name of the :class:`.DirectivityMonitor` to use for calculating far fields.
+            If None, uses the first monitor in `radiation_monitors`.
+
+        Returns
+        -------
+        AntennaParametersData
+            Container with antenna parameters including directivity, gain, and radiation efficiency,
+            computed from the superposition of fields from all excited ports.
+        """
+        # Use the first port as default if none specified
+        if port_amplitudes is None:
+            port_amplitudes = {self.ports[0].name: None}
+        port_names = [port.name for port in self.ports]
+        # Check port names, and create map from port to amplitude
+        port_dict = {}
+        for key in port_amplitudes.keys():
+            port = self.get_port_by_name(port_name=key)
+            port_dict[port] = port_amplitudes[key]
+        # Get the radiation monitor, use first as default
+        # if none specified
+        if monitor_name is None:
+            rad_mon = self.radiation_monitors[0]
+        else:
+            rad_mon = self.get_radiation_monitor_by_name(monitor_name)
+
+        # Create data arrays for holding the superposition of all port power wave amplitudes
+        f = list(rad_mon.freqs)
+        coords = dict(f=f, port=port_names)
+        a_sum = PortDataArray(np.zeros((len(f), len(port_names)), dtype=complex), coords=coords)
+        b_sum = a_sum.copy()
+        # Retrieve associated simulation data
+        combined_directivity_data = None
+        for port, amplitude in port_dict.items():
+            sim_data_port = self.batch_data[self._task_name(port=port)]
+            radiation_data = sim_data_port[rad_mon.name]
+
+            a, b = self.compute_power_wave_amplitudes_at_each_port(
+                self.port_reference_impedances, sim_data_port
+            )
+            # Select a possible subset of frequencies
+            a = a.sel(f=f)
+            b = b.sel(f=f)
+            a_raw = a.sel(port=port.name)
+
+            if amplitude is None:
+                directivity_data = sim_data_port[rad_mon.name]
+                scale_factor = 1.0
+            else:
+                directivity_data = self._monitor_data_at_port_amplitude(
```

Comment:
probably clearer to call it `scaled_directivity_data`, to indicate that they are scaled by scale_factor/a_raw.